src/coreclr/vm/encee.cpp

```diff
@@ -835,7 +835,14 @@ NOINLINE void EditAndContinueModule::FixContextAndResume(
 #if defined(TARGET_X86)
     ResumeAtJit(pContext, oldSP);
 #elif defined(TARGET_WINDOWS) && defined(TARGET_AMD64)
-    ClrRestoreNonvolatileContextWorker(pContext, ssp);
+    if (g_pDebugInterface->IsOutOfProcessSetContextEnabled())
+    {
+        g_pDebugInterface->SendSetThreadContextNeeded(pContext);
```

Comment:
Have you hit a problem where the ClrRestoreNonvolatileContextWorker was not working here? I have thought it would be sufficient as it doesn't use any Windows APIs for restoring the context and thus we should be able to restore the context to any target Rip.

Reply:
Yes this AV's in the EnC code path.

Reply:
That's strange then. Would you mind getting me details on the AV (stack trace, ...)? I would like to understand what exactly was wrong.
With your change, could you also please remove the ssp locals from this function, as it is not needed anymore (it is needed only for CET enabled and in that case, we use this new added code)? You'd also need to change the `ClrRestoreNonvolatileContextWorker(pContext, ssp);` to `ClrRestoreNonvolatileContext(pContext);`

Reply:
> Would you mind getting me details on the AV (stack trace, ...)? I

The stack is empty and RIP is 0.  Further investigation will be needed to understand why ClrRestoreNonvolatileContext does not work in this scenario.  I'll move forward with the existing change for .NET 7 while we continue to investigate this issue.